pkg/service/regression/regression.go

```diff
@@ -81,11 +83,16 @@ func (r *Regression) GetApps(ctx context.Context, cid string) ([]string, error)
 	return r.tdb.GetApps(ctx, cid)
 }
 
-func (r *Regression) Get(ctx context.Context, cid, appID, id string) (models.TestCase, error) {
+func sanitiseInput(s string) string {
+	re := regexp.MustCompile(`(\n|\n\r|\r\n|\r)`)
+	return html.EscapeString(string(re.ReplaceAll([]byte(s), []byte(" "))))
+}
 
+func (r *Regression) Get(ctx context.Context, cid, appID, id string) (models.TestCase, error) {
 	tcs, err := r.tdb.Get(ctx, cid, id)
 	if err != nil {
-		r.log.Error("failed to get testcases from the DB", zap.String("cid", cid), zap.String("appID", appID), zap.Error(err))
+		sanitizedAppID := sanitiseInput(appID)
+		r.log.Error("failed to get testcases from the DB", zap.String("cid", cid), zap.String("appID", sanitizedAppID), zap.Error(err))
```

Comment:
Check failure on line 95 in pkg/service/regression/regression.go

**Log entries created from user input**
This log write receives unsanitized user input from
https://github.com/keploy/keploy/blob/7206b0797938cc1119158a348da6bba8b0856708/http/regression/regression.go#L129-L129

Reply:
@nehagup I might be totally be missing something here, but could this be a false positive? The `app` query is passed as the `appID` param to `regression.Get()` which we're now sanitizing before passing to the logger. I believe this should have fixed this warning.

Reply:
@ditsuke can you please correct the `sanitiseInput` methods replace to replace empty string (eg: `""`) rather than a space( eg: `" "`). Maybe that fixes it?

Reply:
@slayerjain giving it a try, fingers crossed.

Reply:
Thanks, that worked fine! I believe that was a false positive indeed. 😄